src/dom-manipulation.ts

```diff
@@ -19,109 +20,112 @@ export function appendFlightInformationToFlightInfoContainer(flight) {
-        flight[9] ?? 0
+        flight.velocity ?? 0
       )}km/h</span>
       <span class="hidden md:contents">${calculateDirection(
-        flight[10] ?? 0
+        flight.true_track ?? 0
       )}</span>
-      <span>${flight[10] ?? 0}°</span>
-      <span class="hidden lg:contents">${flight[7] ?? 0.0}m</span>
+      <span>${flight.true_track ?? 0}°</span>
+      <span class="hidden lg:contents">${flight.baro_altitude ?? 0.0}m</span>
       <button id="${
-        flight[0] + (flight[1] ?? 0)
+        flight.icao24 + (flight.callsign ?? 0)
       }" class="track-button uppercase">track</button> 
     `;
   }
 }
 
-export function removeOldOutOfScopeFlightInfoRow(inScopeFlightCodes) {
+export function removeOldOutOfScopeFlightInfoRow(inScopeFlightCodes: string[]) {
   const flightInfoDivs = document.querySelectorAll(".single-flight");
   flightInfoDivs.forEach((flightRow) => {
     const inScope = inScopeFlightCodes.includes(flightRow.id);
     if (!inScope) {
       const singleFlightToRemove = document.getElementById(flightRow.id);
-      singleFlightToRemove.remove();
+      singleFlightToRemove!.remove();
     }
   });
 }
 
-export function addEventListenerToFlightInfoButtons(flights) {
+export function addEventListenerToFlightInfoButtons(flights: IFlights[]) {
   const viewButtons = document.querySelectorAll(".track-button");
   viewButtons.forEach((button) => {
     const relInfo = flights.find(
-      (flight) => flight[0] + flight[1] === button.id
+      (flight) => flight.icao24 + flight.callsign === button.id
     );
     if (relInfo) {
       button.addEventListener("click", () => {
-        toggleFlightFocus(event, relInfo);
+        toggleFlightFocus(event as MouseEvent, relInfo);
       });
     }
   });
 }
 
 export async function minimiseLoadingScreen() {
   const loadScreen = document.getElementById("loading");
-  loadScreen.classList.add("animate-loadAnime");
+  loadScreen!.classList.add("animate-loadAnime");
 }
 
-function toggleFlightFocus(event, fltInfo) {
+function toggleFlightFocus(event: MouseEvent, fltInfo: IFlights) {
   const mapElement = document.getElementById("map");
-  const button = event.target;
+  const button = event.target as HTMLElement;
 
-  if (button.innerText === "CLOSE") {
+  if (button!.innerText === "CLOSE") {
     // If button has already been clicked
-    mapElement.style.visibility = "hidden";
+    mapElement!.style.visibility = "hidden";
     button.innerText = "track";
     resetMapLocationView();
   } else {
     // if a button is clicked
-    mapElement.style.visibility = "visible";
-    button.innerText = "close";
-    setMapAndMarkerToCurrentFlightLocation(fltInfo[6], fltInfo[5], fltInfo[10]);
+    mapElement!.style.visibility = "visible";
+    button!.innerText = "close";
+    setMapAndMarkerToCurrentFlightLocation(fltInfo.latitude, fltInfo.longitude, fltInfo.true_track);
   }
-  showAndHideButtonsAfterClick(button.innerText);
+  showAndHideButtonsAfterClick(button!.innerText);
```

Comment:
Be careful of the bang operator. Use null checks instead if possible

Reply:
I reworded the button bang operator. The bang I used below this is for an event that will exist without fail same with the parent element.